isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+
+Zu intensive Nutzung kann leicht zu einem schwer durchschaubaren Programmfluss führen.
+Deshalb soll AOP nur für folgende Bereichen genutzt werden: die Steuerung von Transaktionen, die Überwachung und die Berechtigungsprüfung.
+
+Explizit nicht benutzt werden soll AOP für die Fehlerbehandlung.
+
+Die Verwendung von AOP für andere Bereiche ist nur in begründeten Ausnahmefällen erlaubt.
+
+[[aop-fuer-transaktionssteuerung-verwenden]]
+=== AOP für Transaktionssteuerung verwenden
+
+Für die Transaktionssteuerung ist Spring-AOP mit den dafür vorgesehenen Klassen von Spring einzusetzen.
+Die Umsetzung geschieht im xref:software-technisch/backend/anwendungskern.adoc#transaktionssteuerung[Anwendungskern] oder in der xref:blaupausen:detailkonzept-komponente-service/master.adoc#transaktionssteuerung[Serviceschicht].
+Zusammengefasst gilt:
+
+* Instrumentiert werden alle Schnittstellenmethoden des Anwendungskerns.
+* Für jeden Aufruf des Anwendungskerns wird eine Transaktion gestartet.
+* Falls kein Fehler auftritt, wird die Transaktion abgeschlossen (Commit), sonst zurückgerollt (Rollback).
+
+[[aop-fuer-berechtigungspruefungen-verwenden]]
+=== AOP für Berechtigungsprüfungen verwenden
+
+Die Berechtigungsprüfung wird über Spring-AOP mit den vom xref:isy-security:nutzungsvorgaben/master.adoc[Baustein Security] angebotenen Annotationen umgesetzt.
+
+[[aop-nicht-fuer-das-logging-von-exceptions-verwenden]]
+=== AOP nicht für das Logging von Exceptions verwenden
+
+Sämtliche in einem Backend geworfenen und nicht behandelten Ausnahmen müssen inklusive ihrer Stack-Traces geloggt werden.
+Geloggt wird dies in den Methoden der Schnittstellen-Beans.
+Hierfür soll Spring-AOP nicht verwendet werden.
+
+Schnittstellen-Beans transformieren die Geschäftsobjekte des Anwendungskerns in Transportobjekte der Schnittstelle.
+Die Stack-Traces der Exceptions werden dabei nicht übertragen, da diese internen Informationen dem Aufrufer keinen Mehrwert bieten.
+Für das Logging des Backends selbst sind sie jedoch wertvoll.
+Statt AOP ist das im xref:blaupausen:detailkonzept-komponente-service/master.adoc[Detailkonzept Komponente Service] beschriebene Konstrukt der Exception-Fassade zu verwenden.
+
+[[dependency-injection]]
+== Dependency Injection
+
+Um im Anwendungskern das Single-Responsibility-Prinzip nicht zu verletzten, wird auf das Entwurfsmuster _Dependency Injection_ zurückgegriffen.
+Dabei werden die Verantwortlichkeiten für den Aufbau des Abhängigkeitsnetzes zwischen den Objekten aus den einzelnen Klassen in eine zentrale Komponente überführt.
+Anders als bei der herkömmlichen Vorgehensweise in der objektorientierten Programmierung ist bei der _Dependency Injection_ nicht jedes Objekt selbst dafür zuständig, seine Abhängigkeiten (benötigte Objekte und Ressourcen) zu verwalten.
+
+In der <<image-dependency-injection,folgenden Grafik>> ist das Entwurfsmuster der _Dependency Injection_ exemplarisch visualisiert.
+
+[[image-dependency-injection]]
+.Dependency Injection
```

Comment:
Geht es in diesem Kapitel um Vorgaben?
Wenn ja, dann sollte dieses Kapitel in "Vorgaben zur  Dependency Injection" umbenannt werden.